graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/QueryValidationServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.validation;
+
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.graylog.plugins.views.search.ParameterProvider;
+import org.graylog.plugins.views.search.Query;
+import org.graylog.plugins.views.search.elasticsearch.QueryStringDecorators;
+import org.graylog.plugins.views.search.rest.MappedFieldTypeDTO;
+import org.graylog2.indexer.fieldtypes.MappedFieldTypesService;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Singleton
+public class QueryValidationServiceImpl implements QueryValidationService {
+
+    private final LuceneQueryParser luceneQueryParser;
+    private final MappedFieldTypesService mappedFieldTypesService;
+    private final QueryStringDecorators queryStringDecorators;
+
+    @Inject
+    public QueryValidationServiceImpl(LuceneQueryParser luceneQueryParser, MappedFieldTypesService mappedFieldTypesService, QueryStringDecorators queryStringDecorators) {
+        this.luceneQueryParser = luceneQueryParser;
+        this.mappedFieldTypesService = mappedFieldTypesService;
+        this.queryStringDecorators = queryStringDecorators;
+    }
+
+    @Override
+    public ValidationResponse validate(ValidationRequest req) {
+        final String query = decoratedQuery(req);
+        try {
+            final ParsedQuery parsedQuery = luceneQueryParser.parse(query);
+            final List<ParsedTerm> unknownFields = getUnknownFields(req, parsedQuery);
+            final List<ParsedTerm> invalidOperators = parsedQuery.invalidOperators();
+            final List<ValidationMessage> explanations = getExplanations(unknownFields, invalidOperators);
+            final ValidationStatus status = explanations.isEmpty() ? ValidationStatus.OK : ValidationStatus.WARNING;
```

Comment:
compare to the DTO that only copies explanations if status != OK

Reply:
fixed, thx!